modules/ocf/manifests/networking.pp

```diff
@@ -12,61 +13,75 @@
 
   if size($nameservers) > 3 {
     fail("Can't have more than 3 nameservers")
   }
 
+  $fqdn = $::clientcert
+  $hostname = regsubst($::clientcert, '^([\w-]+)\..*$', '\1')
+  $linked_ifaces_array = split($::ifaces_linked, ' ')
+  $first_active_iface = $linked_ifaces_array[0]
+
   # packages
-  if $bridge {
-    package { 'bridge-utils': }
+  if $bond {
+    package { 'ifenslave': }
   }
 
   package { 'resolvconf':
     ensure => purged,
   }
 
-  if $::lsbdistcodename == 'jessie' {
-    if $bridge {
-      $iface = 'br0'
-      $br_iface = 'eth0'
+  # $logical_primary_interface is the one we want to add the IP address to.
+  # in the basic case, it's just the active physical interface (desktops, VMs)
+  # it could also be a bridge interface or bond interface with one or more
+  # physical interfaces slaved to it.
+
+  if $bridge {
+    $logical_primary_interface = 'br0'
+
+    if $bond {
+      $bridged_iface = 'bond0'
     } else {
-      $iface = 'eth0'
+      $bridged_iface = $first_active_iface
     }
+  } elsif $bond {
+    $logical_primary_interface = 'bond0'
   } else {
-    if $::lsbdistid == 'Raspbian' {
-      # The raspberry pi has wifi, so we use that for networking
-      $ifaces_array = split($::interfaces, ',')
-      $br_iface = grep($ifaces_array, 'wl.+')[0]
-    } else {
-      $br_iface = $::iface_linked
+    $logical_primary_interface = $first_active_iface
+  }
+
+  if $::lsbdistid == 'Raspbian' {
+    # The raspberry pi has wifi, so we use that for networking
+    $logical_primary_interface = 'wlan0'
+  }
+
+  if $bridge {
+    file { '/etc/network/interfaces.d/br0':
+      content => template('ocf/networking/interface_bridge.erb');
     }
+  }
 
-    # If using bridged networking, use the interface found above as the
-    # interface being bridged to.
-    if $bridge {
-      $iface = 'br0'
-    } else {
-      if $br_iface and ($br_iface != '') {
-        $iface = $br_iface
-      } else {
-        # This is the default ethernet interface name on new VMs. While not
-        # necessarily correct, falling back will help make sure VMs upgraded
-        # from jessie have networking configured when they reboot, since they
-        # retain the old 'eth0' interface name until after reboot.
-        #
-        # TODO: find out how to predict new interface names
-        $iface = 'ens3'
-      }
+  if $bond {
+    file { '/etc/network/interfaces.d/bond0':
+      content => template('ocf/networking/interface_bond.erb');
+    }
+  }
+
+  unless ($bond or $bridge) {
+    file { "/etc/network/interfaces.d/${logical_primary_interface}":
+      content => template('ocf/networking/interface_normal.erb');
     }
   }
 
   # network configuration
   file {
     '/etc/network/interfaces':
-      content => template('ocf/networking/interfaces.erb');
+      source => 'puppet:///modules/ocf/networking/interfaces';
+    '/etc/network/interfaces.d':
+      ensure  => directory,
+      recurse => true,
+      purge   => true;
```

Comment:
Are you sure `purge` will work here? The files aren't actually in the directory, they get filled in by templates elsewhere. Hopefully this won't delete then re-add them on each puppet run.

Reply:
the docs mention that `purge` will remove all unmanaged files. in my testing it doesn't appear to delete and re-add so I assume it's intelligent enough to not recreate them every time.